### PR TITLE
Feature/13569 jo

### PR DESCRIPTION
**Pull request by jeffrey-opdam**
Thank you for your pull request!
By completing this pull request, you agree to the [Contributing License Agreement](https://github.com/ALM-Rangers/VSTS-Bot/blob/master/.github/CLA.md).

Fixes # .

Changes proposed in this pull request:  
- User can now create a subscription for their approvals.
- 
- 

@ALM-Rangers/VSTS-Bot

### CODE REVIEW

**Comment by jeffrey-opdam** [comment]
Can you guys have good look at it? Make sure everything makes sense.

flow starts with typing 'subscriptions'.

File: src/VSTS-Bot.Api/Controllers/EventController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                        UriFactory.CreateDocumentCollectionUri("botdb", "subscriptioncollection"), querySpec)
+                    .ToList();
+
+                var tasks = new List<Task>();
+
+                foreach (var subscription in subscriptions)
+                {
+                    foreach (var strategy in this.strategies.Where(s => s.ShouldHandle(@event)))
+                    {
+                        var t = strategy.Handle(@event, subscription);
```
Comment (jeffrey-opdam) [inline src/VSTS-Bot.Api/Controllers/EventController.cs]: Of loading to a strategy. Meaning for each EventType we create an strategy, which contains the logic what to do with that event.

File: src/VSTS-Bot.Api/DI/CommandMetadataAttribute.cs [hunk 2]
```diff
@@ -22,15 +23,15 @@ public sealed class CommandMetadataAttribute : Attribute
         /// <summary>
         /// Initializes a new instance of the <see cref="CommandMetadataAttribute"/> class.
         /// </summary>
-        /// <param name="command">The actual command.</param>
-        public CommandMetadataAttribute(string command)
+        /// <param name="commands">The actual commands.</param>
+        public CommandMetadataAttribute(params string[] commands)
```
Comment (jeffrey-opdam) [inline src/VSTS-Bot.Api/DI/CommandMetadataAttribute.cs]: An dialog sometimes need be triggered by multiple commands.

File: src/VSTS-Bot.Api/Dialogs/SubscriptionsDialog.cs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                    IsActive = true,
+                    ProfileId = this.Profile.Id,
+                    RecipientId = activity.From.Id,
+                    RecipientName = activity.From.Name,
+                    ServiceUri = new Uri(activity.ServiceUrl),
+                    SubscriptionType = subscriptionType,
+                    UserId = activity.From.Id
+                };
+
+                var strategy = this.strategies.First(s => s.CanGetSubscription(subscriptionType));
```
Comment (jeffrey-opdam) [inline src/VSTS-Bot.Api/Dialogs/SubscriptionsDialog.cs]: Of loading to a strategy, which contains the logic what details needs to be in the subscription.

**Comment by jeffrey-opdam** [comment]
When testing from your local machine. Make sure to replace the url with the ngrok url in the subscription of the service hook in vsts.

File: src/VSTS-Bot.Api/Controllers/EventController.cs (hunk 1 above)
Comment (dkmsft) [reply src/VSTS-Bot.Api/Controllers/EventController.cs]: very nice approach

**Pull request merged**